node-repository/src/main/java/com/yahoo/vespa/hosted/provision/autoscale/Autoscaler.java

```diff
@@ -142,6 +143,8 @@ private boolean similar(double r1, double r2, double threshold) {
             Optional<Flavor> bestFlavor = Optional.empty();
             for (Flavor flavor : nodeRepository.getAvailableFlavors().getFlavors()) {
                 if ( ! flavor.resources().satisfies(resources.nodeResources())) continue;
```

Comment:
`flavor.resources()` returns the real resources, need to run this through `hostResourcesCalculator` so the `bestFlavor` ends up as an advertised flavor.

Reply:
> You should also do nodeResourceLimits.enlargeToLegal [...]

Only if we add host flavors that cannot be allocated, but ok.

Reply:
We already have (a few `t3-` flavors)